File: ws/sanchez/u3/ws29Singleton/Singleton/src/main/java/ec/edu/espe/singleton/Tax.java
package ec.edu.espe.singleton;

/**
 *
 * @author Joel Sanchez
 */
public class Tax {

    private static Tax instance;
    private float percentage;

    private Tax() {
        percentage = 0.0f;
    }

    public static Tax getInstance() {
        if (instance == null) {
            instance = new Tax();
        }
        return instance;
    }

    public void updateTaxPercentage(float percentage) {
        this.percentage = percentage;
    }

    public float getPercentage() {
        return percentage;
    }

    public float salesTotal(float sale) {
        return sale + (sale * percentage / 100);
    }
}

File: hw/tipantiza/u3/hw23SingletonPattern/Singleton_py/tax.py
class Tax:
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(Tax, cls).__new__(cls)
            cls._instance._percentage = 0.0
        return cls._instance

    def update_tax_percentage(self, percentage):
        self._percentage = percentage

    def get_percentage(self):
        return self._percentage

    def sales_total(self, sale):
        return sale + (sale * self._percentage / 100)